### PR TITLE
chore: add dependabot config

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
N/A

*Description of changes:*
This PR adds explicit configuration for dependabot for Go dependency and GitHub Actions dependency updates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by austinvazquez** [comment]
Not sure if this is related but the dependabot commit messages since https://github.com/aws/shim-loggers-for-containerd/commit/ff437666a76f0131bdf6076e52aaf85161cff624 are just unwieldy. Not sure what changed but having explicit config will help track it.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+updates:
+  # Dependencies listed in go.mod
+  - package-ecosystem: "gomod"
+    directory: "/"  # Location of package manifests
+    schedule:
+      interval: "daily"
+    ignore:
+      # support LTS 1.6.x containerd, ignore dependabot alerts for 1.7
+      - dependency-name: "github.com/containerd/containerd"
+        versions: ["1.7.x"]
```
Comment (ningziwen) [inline .github/dependabot.yml:14]: Is it necessary to ignore it here? Maybe we can let it cut PR but don't merge it?

Comment (austinvazquez) [reply .github/dependabot.yml]: Update here. containerd 1.7.x has extended support now to align with containerd 1.6 LTS. So updates should be fine here.

**Review by BinSquare — APPROVED** [review]
lgtm

**Pull request merged**